src/System.Private.CoreLib/shared/System/Globalization/CompareInfo.cs

```diff
@@ -977,7 +977,7 @@ public unsafe virtual int IndexOf(string source, char value, int startIndex, int
             if ((options & ValidIndexMaskOffFlags) != 0 && (options != CompareOptions.Ordinal && options != CompareOptions.OrdinalIgnoreCase))
                 throw new ArgumentException(SR.Argument_InvalidFlag, nameof(options));
 
```

Comment:
It would be even better to refactor this so that this allocation is not required at all, but this is still an incremental improvement.